test/object-store/sync/session/progress_notifications.cpp

```diff
@@ -395,90 +383,245 @@ TEST_CASE("progress notification", "[sync][session][progress]") {
 
-        register_default_upload_callback();
-        REQUIRE_FALSE(callback_was_called);
+        SECTION("for upload notifications") {
+            // Prime the progress updater
+            current_transferred = 60;
+            current_transferrable = 501;
+            const uint64_t original_transferrable = current_transferrable;
+            double current_estimate = current_transferred / double(current_transferrable);
+            progress.update(21, 26, current_transferred, current_transferrable, 1, 21 / double(26), current_estimate,
+                            0);
 
-        uint64_t current_transferred = 66;
-        uint64_t current_transferrable = 582;
-        progress.update_upload(current_transferred, current_transferrable, 3);
-        REQUIRE_FALSE(callback_was_called);
+            progress.register_callback(
+                [&](auto xferred, auto xferable, double ep) {
+                    transferred = xferred;
+                    transferrable = xferable;
+                    upload_estimate = ep;
+                    callback_was_called = true;
+                },
+                NotifierType::upload, false, 0);
+            // Wait for the initial callback.
+            REQUIRE(callback_was_called);
 
-        current_transferred = 77;
-        current_transferrable = 1021;
-        progress.update(0, 0, current_transferred, current_transferrable, 4);
-        REQUIRE(callback_was_called);
-        CHECK(transferred == current_transferred);
-        // should not have captured transferrable from the first update
-        CHECK(transferrable == current_transferrable);
-    }
+            // Now manually call the notifier handler a few times.
+            callback_was_called = false;
+            current_transferred = 66;
+            current_transferrable = 582;
+            current_estimate = current_transferred / double(current_transferrable);
+            progress.update(21, 26, current_transferred, current_transferrable, 1, 21 / double(26), current_estimate,
+                            0);
+            CHECK(callback_was_called);
+            CHECK(transferred == current_transferred);
+            CHECK(transferrable == original_transferrable);
+            CHECK(upload_estimate == current_transferred / double(original_transferrable));
+
+            // Second callback
+            callback_was_called = false;
+            current_transferred = original_transferrable + 100;
+            current_transferrable = 1021;
+            current_estimate = current_transferred / double(current_transferrable);
+            progress.update(68, 191, current_transferred, current_transferrable, 1, 68 / double(191),
+                            current_estimate, 0);
+            CHECK(callback_was_called);
+            CHECK(transferred == current_transferred);
+            CHECK(transferrable == original_transferrable);
+            CHECK(upload_estimate == std::min(1.0, current_transferred / double(original_transferrable)));
```

Comment:
i've updated this to just `CHECK(upload_estimate == 1.0)`since the sync session notification won't let you have an estimate greater than 1.0 for non-streaming upload notifications